src/controllers/mainController.ts

```diff
@@ -717,9 +717,18 @@ export default class MainController implements vscode.Disposable {
                 // make a new connection
                 connectionCreds.database = databaseName;
                 if (!this.connectionManager.isConnecting(nodeUri)) {
-                    const promise = new Deferred<boolean>();
```

Comment:
Removed this bad pattern of creating and waiting for a custom promise to figure out connection creation was successful.